pkg/grid/src/app.tsx

```diff
@@ -10,23 +12,79 @@ import useContactState from './state/contact';
 import api from './state/api';
 import { useMedia } from './logic/useMedia';
 import { useHarkStore } from './state/hark';
-import { useSettingsState, useTheme } from './state/settings';
-import { useLocalState } from './state/local';
+import { useSettingsState, useBrowserSettings, useTheme } from './state/settings';
+import { useBrowserId, useLocalState } from './state/local';
 import { ErrorAlert } from './components/ErrorAlert';
 import { useErrorHandler } from './logic/useErrorHandler';
 
 const getNoteRedirect = (path: string) => {
   if (path.startsWith('/desk/')) {
     const [, , desk] = path.split('/');
-    return `/app/${desk}`;
+    return `/apps/${desk}`;
+  }
+
+  if (path.startsWith('/grid/')) {
+    // Handle links to grid features (preferences, etc)
+    const route = path
+      .split('/')
+      .filter((el) => el !== 'grid')
+      .join('/');
+    return route;
   }
   return '';
 };
 
+const getId = async () => {
+  const fpPromise = FingerprintJS.load();
+  const fp = await fpPromise;
+  const result = await fp.get();
+  return result.visitorId;
+};
+
 const AppRoutes = () => {
   const { push } = useHistory();
   const { search } = useLocation();
   const handleError = useErrorHandler();
+  const browserId = useBrowserId();
+  const settings = useBrowserSettings();
+
+  useEffect(() => {
+    getId().then((value) => {
+      useLocalState.setState({ browserId: value });
+    });
+  }, [browserId]);
+
+  useEffect(() => {
+    // Check if user has previously stored settings for this browser.
+    // If not, send a notification prompting them to do so.
+    // Set both settings to false so that they're not bugged again.
+    if (browserId !== '') {
+      const thisBrowserSettings = settings.filter((el: any) => el.browserId === browserId)[0];
+      if (!thisBrowserSettings) {
+        api.poke(
+          addNote(
+            {
+              path: '/browser-settings',
+              place: { desk: 'garden', path: '/desk/garden' }
+            },
+            {
+              title: [{ text: 'Browser Preferences' }],
+              time: Date.now(),
+              content: [
+                { text: "You haven't set your preferences for this browser, set them now?" }
```

Comment:
We should get @urcades / @e-shee to review this language

Reply:
Looks good.